cmd/root.go

```diff
@@ -151,4 +157,5 @@ func addSharedFlags(cmd *cobra.Command) {
 	cmd.Flags().StringVarP(&types, "types", "t", "apt", "This flag sets the list of analyzer types to use.  It expects a comma separated list of supported analyzers.")
 	cmd.Flags().BoolVarP(&save, "save", "s", false, "Set this flag to save rather than remove the final image filesystems on exit.")
 	cmd.Flags().BoolVarP(&util.SortSize, "order", "o", false, "Set this flag to sort any file/package results by descending size. Otherwise, they will be sorted by name.")
+	cmd.Flags().StringVarP(&filename, "filename", "f", "", "View the diff of a file in both containers (can only be used with container-diff diff)")
```

Comment:
You should add this flag to the diff subcommand specific flags instead of have it be at the root level.  That will let you remove a lot of the flag validation.